matrix-synapse.tools.synadm: 0.40 -> 0.41.2

mweinelt opened the pull request:
https://github.com/JOJ0/synadm/releases/tag/v0.41
https://github.com/JOJ0/synadm/releases/tag/v0.41.1
https://github.com/JOJ0/synadm/releases/tag/v0.41.2

###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

mweinelt committed:
matrix-synapse.tools.synadm: 0.40 -> 0.41.2

https://github.com/JOJ0/synadm/releases/tag/v0.41
https://github.com/JOJ0/synadm/releases/tag/v0.41.1
https://github.com/JOJ0/synadm/releases/tag/v0.41.2

emilylange reviewed (approved):
Validated (extremely basic) functionality via `result/bin/synadm version` on `x86_64-linux` :+1: 

PS: Considering there has been some time between the release and your bump (not judging!) and you are the only maintainer, you might be interested in adding `nix-update-script` so @r-ryantm can try to pick this up in the future:

```diff
diff --git a/pkgs/servers/matrix-synapse/tools/synadm.nix b/pkgs/servers/matrix-synapse/tools/synadm.nix
index 77117a674ce..91efac65b60 100644
--- a/pkgs/servers/matrix-synapse/tools/synadm.nix
+++ b/pkgs/servers/matrix-synapse/tools/synadm.nix
@@ -1,4 +1,5 @@
 { lib
+, nix-update-script
 , python3
 }:
 
@@ -28,6 +29,8 @@ python3.pkgs.buildPythonApplication rec {
     runHook postCheck
   '';
 
+  passthru.updateScript = nix-update-script { };
+
   meta = with lib; {
     description = "Command line admin tool for Synapse";
     longDescription = ''
```

(Verified this works by going back one commit and got the same diff/hash)

https://r.ryantm.com/log/matrix-synapse-tools.synadm/2023-04-26.log

IndeedNotJames commented:
@ofborg build matrix-synapse-tools.synadm

emilylange committed:
matrix-synapse.tools.synadm: Add update script

The r-ryantm bot tries to update the package, but fails to do so, unless
we provide an update script.

Co-Authored-By: Martin Weinelt <hexa@darmstadt.ccc.de>
